pkg/common-controller/snapshot_controller.go

```diff
@@ -170,17 +170,14 @@ func (ctrl *csiSnapshotCommonController) syncSnapshot(snapshot *crdv1.VolumeSnap
 	klog.V(5).Infof("syncSnapshot [%s]: check if we should remove finalizer on snapshot PVC source and remove it if we can", utils.SnapshotKey(snapshot))
 
 	// Check if we should remove finalizer on PVC and remove it if we can.
-	if err := ctrl.checkandRemovePVCFinalizer(snapshot); err != nil {
+	if err := ctrl.checkandRemovePVCFinalizer(snapshot, false); err != nil {
```

Comment:
IMHO, checkandRemovePVCFinalizer *can* take the responsibility to handle "snapshot.ObjectMeta.DeletionTimestamp != nil" case,
and the bool flag can be saved?

Reply:
I think it is safer to have this bool flag passed in.  Even when "snapshot.ObjectMeta.DeletionTimestamp != nil", it is possible that snapshot is also used as a source for another PVC.  In that case, we still want to keep PVC finalizer on the snapshot's source if snapshot is not ready yet.